index.js

```diff
@@ -77,3 +93,98 @@ function formatAttachments(attachments = []){
     return attachments.map(attachment => attachment.url)
         .join('\n');
 }
+
+function rollDecider(command){
+    const commandArray = command.toLowerCase().split('');
+    const allowableCharacters = ['1','2','3','4','5','6','7','8','9','0','d','r',' ','+','-'];
+    const commandCharacters = ['d','r','+','-']
+
+    commandCharacters.forEach( character => {
+        let tempIndex = commandArray.indexOf(character);
+        if ( tempIndex > 0 ){
+            if ( commandArray.indexOf(character, (tempIndex + 1)) > 0 ) {
+                throw 'You can have one dice, modifier, or operator at a time'
+            }
+        }
+        if ((character === 'r') && (tempIndex != -1) && ((tempIndex + 1) < (commandArray.length))) {
+            throw '"r" should only be at the end of your command'
+        }
+    })
+
+    commandArray.forEach( character => {
+        if ( !allowableCharacters.includes(character) ){
+            throw 'Please remove invalid characters'
+        }
+    })
+
+    const commandCleaned = command.toLowerCase().split(' ').join('').split('r').join(''); // removes spaces and Rs 
```

Comment:
I know that you don't want to use regex, but `command.toLowerCase().replace(/ /g, '').replace('r', '')` reads a lot better to me than splitting and rejoining.

Reply:
fair, I will probably circle back around when I can make regex fit into my brain.